### DISCUSSION TITLE
Upgrade to HACS version

### DISCUSSION THREAD

**Discussion by Paddy-Wa** [ExperienceLovelace/ha-floorplan#83] [Q&A] [answered]
I successfully implemented floorplan via the older floorplan-card.js implementation. Now that this has moved to HACS I would like to take advantage of any updates and such. To date, each time I attempt to implement with my existing SVG and CSS, nothing actually changes color anymore. Is there a recommended way to do that update to ensure functionality remains?

**Comment by OzGav** [comment]
It might be because you have a grouped object. To make this work now you just need to add a new bit to your CSS. Whatever the CSS class is called just add another one with the same name followed by an asterisk as follows:

```
.lamp-off, .lamp-off *  {
  stroke: #000000 !important;
  opacity: 0.5 !important;
  stroke-opacity: 0 !important;
}
```

**Comment by exetico** [comment, marked as answer]
Hi,

We don't have any migration-guide, or something simular. @pkozul are still working on the "options"-page in our official docs. I guess it's a good idea, to have a short/simple guide on how to migrate most things to the new solution.

Quite a few things has been changed, and done much better :-) 

For a color-change example, have you seen our "Light"-example?
https://experiencelovelace.github.io/ha-floorplan/docs/example-light/

**Reply by Paddy-Wa** [reply]
Thanks for the Example. I have all the color changing fancyness working under the previous install of ha-floorplan (lovelace card).
But when I switch to the HACS version the colors stop working and the SVG stays whatever colors it started with. It's worth noting that I am able to click the items on the SVG and get the "more-info" dialog for them. So it's really just the CSS coloring that seems to be amiss for me.

**Reply by exetico** [reply]
No problem. A good starting-point would be to share your configs and more. As mentioned, many logics has been changed 😊 (if you're talking about a move from the old ha-floorplan from @pkozul 's repo.)

**Reply by OzGav** [reply]
> I am able to click the items on the SVG and get the "more-info" dialog for them. So it's really just the CSS coloring that seems to be amiss for me.

That sounds to me like perhaps you haven't changed the rule to the new standard for responding to the click/touch (because if the entity doesn't change state I am not sure how it will change colour - unless it is changing state via another means).
Previously you would had something like:
```
            - action:
                service: light.toggle
```

Now you should have:
```
            - tap_action:
                service: light.toggle
```

If you add the log_level option to you ui-lovelace.yaml you might see some useful error messages?
```
  - panel: true
    cards:
      - config:
          log_level: info
          image:
            location: .....
```

**Reply by OzGav** [reply]
Hi @paddy0furniture just checking if your problem has been resolved?

**Reply by bbostock** [reply]
I have been using the old ha-floorplan from @pkozul 's repo very successfully and have built up a very complex floor plan since installation. Trying to convert is driving me nuts without a guide to what has changed. Trying to work it out from examples is very frustrating. Some things work, and others don't. But is it me or the new code? What I find most frustrating is that the same type is used ('custom:floorplan-card') so I can't easily migrate AND retain the old version whilst I work out the new version. I am giving up now, and I'll wait for the guides to be completed.

**Reply by Paddy-Wa** [reply]
> Hi @paddy0furniture just checking if your problem has been resolved?

It has not but the old version is working great and it's fairly complex now.
Really odd it all seems to work except for any colors from the CSS. 
I will likely wait and try to implement as new at some point rather than adapting what I have.

**Comment by Paddy-Wa** [comment]
I've not had any luck resolving and have gone back to the older lovelace
implementation for the time being.
I'll probably revisit somewhere down the road...

Best Regards,
*Patrick Donohue*

**Reply by bbostock** [reply]
Ive created a renamed version of the old Lovelace floorpan as floorplan1. So with this installed all the old stuff still works, and you can test the new stuff in your own time. Simple installation notes inside the zip. Modify the original floorplan url resource to be floorplan1 and if the new one is installed both old and new will run together.
[floorplan1.zip](https://github.com/ExperienceLovelace/ha-floorplan/files/6295755/floorplan1.zip)

**Reply by exetico** [reply]
Here's one post with a bit of usefull information:

https://community.home-assistant.io/t/floorplan-now-available-as-a-lovelace-card/115489/327?u=exetico

If you need examples for specific things, kindly ask here.

I'm still waiting for a "options" page from @pkozul - but you'll find a few of the options in the examples located in our docs:
https://experiencelovelace.github.io/ha-floorplan/docs/examples/

**Reply by bbostock** [reply]
Thank you for the information. I am slowly working it out, and I'll come back here if I get stuck and can't figure it out.

**Reply by bbostock** [reply]
I have pretty much everything working now. 

One further question, Is pan/zoom available?

I notice it is commented out in the source code. Could it be added back? 
I use it extensively on the iPhone app with the old floorpan. Or is there an alternative way I could set it up so I can do the same?

**Reply by OzGav** [reply]
I use floorplan on my iPad (amongst other devices) and I find pinch zoom and pan works by default? This is through the safari browser but yes I just tried through the app and it doesn’t work there...

**Reply by exetico** [reply]
I agree with OzGav. What's the actual needs here, which are not already working as expected? :-)

**Reply by bbostock** [reply]
Ok, so as OzGav says on my iPhone/iPad, new floorplan pan/zooms beautifully but not on the HA App, which is what I always use. Do you think this is this an App issue or a new Floorpan issue?

On my MacBook Pro under Big Sur using Safari it doesn't zoom/pan and the animations are all wrong. But it does resize with the scroll wheel.

Also on my MBP using Chrome (my usual HA browser) doesn't zoom/pan or respond to the scroll wheel but the animations are perfect.
Quite confusing.

**Reply by exetico** [reply]
Generally you shouldn't expect the app to support normal browser-like zoom. The content are fitted to the page.

Adding zoom in the ha-floorplan component itself, could be a solution - but I can't personally see the high need of a feature like that 😊.

@pkozul - What do you think?

**Reply by bbostock** [reply]
My full floorpan is quite extensive and so on a phone, zoom is essential. My eyesight is not so good anymore, and the old floorplan had a fantastic zoom-pan option which I relied on.

**Reply by bbostock** [reply]
I went back and looked around the latest Home Assistant iOS app, and it now has an option to enable zoom/pan. 
And the new Floorplan works brilliantly! 
They sneaked that feature in, but much appreciated.

**Reply by exetico** [reply]
That's great to hear! Thank you, for letting us know :-)

**Comment by exetico** [comment]
_This is a copy/paste answer, previously added as a reply to an answer. Please let us know how it's going?_

Here's one post with a bit of usefull information:

https://community.home-assistant.io/t/floorplan-now-available-as-a-lovelace-card/115489/327?u=exetico

If you need examples for specific things, kindly ask here.

I'm still waiting for a "options" page from @pkozul - but you'll find a few of the options in the examples located in our docs:
https://experiencelovelace.github.io/ha-floorplan/docs/examples/

**Reply by bbostock** [reply]
It's all working now, thanks. My only limitation is I don't have zoom/pan on the browsers on my Mac, but that is less of an issue. (But I'm going to have a good look around the HA settings just in casein's hidden there too!)
I've attached a copy of my plan. I have far infrared heating panels, so they are the blue squares on the plan (red when on). All of the panels are individually controlled from a bluetooth thermostat and a Tasmota plug via Node-Red. I can throttle the energy usage with prioritised rooms and times. Works a treat, although I am currently implementing a failsafe backup in case HA hardware crashes. My water tank heater comes on when there is excess solar power or the temperature drops below a threshold.
<img width="928" alt="New Floorplan" src="https://user-images.githubusercontent.com/3266107/114903799-0fb2b480-9e0f-11eb-9356-202e6d52701e.png">

**Reply by exetico** [reply]
Oh, are you @paddy0furniture, too? 😁

**Reply by bbostock** [reply]
No completely different. Sorry if I've caused any confusion.
